qiskit/providers/ibmq/ibmqbackendservice.py

```diff
@@ -0,0 +1,296 @@
+
+        Returns:
+            list(IBMQJob): list of IBMQJob instances
+
+        Raises:
+            IBMQBackendValueError: status keyword value unrecognized
+        """
+        # Build the filter for the query.
+        api_filter = {}
+
+        if backend_name:
+            api_filter['backend.name'] = backend_name
+
+        if status:
+            if isinstance(status, str):
+                status = JobStatus[status]
+            if status == JobStatus.RUNNING:
+                this_filter = {'status': ApiJobStatus.RUNNING.value,
+                               'infoQueue': {'exists': False}}
+            elif status == JobStatus.QUEUED:
+                this_filter = {'status': ApiJobStatus.RUNNING.value,
+                               'infoQueue.status': 'PENDING_IN_QUEUE'}
+            elif status == JobStatus.CANCELLED:
+                this_filter = {'status': ApiJobStatus.CANCELLED.value}
+            elif status == JobStatus.DONE:
+                this_filter = {'status': ApiJobStatus.COMPLETED.value}
+            elif status == JobStatus.ERROR:
+                this_filter = {'status': {'regexp': '^ERROR'}}
+            else:
+                raise IBMQBackendValueError('unrecognized value for "status" keyword '
+                                            'in job filter')
+            api_filter.update(this_filter)
+
+        if job_name:
+            api_filter['name'] = job_name
+
+        if db_filter:
+            # status takes precedence over db_filter for same keys
+            api_filter = {**db_filter, **api_filter}
+
+        # Retrieve the requested number of jobs, using pagination. The API
+        # might limit the number of jobs per request.
+        job_responses = []
+        current_page_limit = limit
+
+        while True:
+            job_page = self._provider._api.list_jobs_statuses(
+                limit=current_page_limit, skip=skip, extra_filter=api_filter)
+            job_responses += job_page
+            skip = skip + len(job_page)
+
+            if not job_page:
+                # Stop if there are no more jobs returned by the API.
+                break
+
+            if limit:
+                if len(job_responses) >= limit:
+                    # Stop if we have reached the limit.
+                    break
+                current_page_limit = limit - len(job_responses)
+            else:
+                current_page_limit = 0
+
+        job_list = []
+        for job_info in job_responses:
+            kwargs = {}
+            try:
+                job_kind = ApiJobKind(job_info.get('kind', None))
+            except ValueError:
+                # Discard pre-qobj jobs.
+                break
+
+            kwargs['use_websockets'] = True
+            if job_kind == ApiJobKind.QOBJECT_STORAGE:
+                kwargs['use_object_storage'] = True
+
+            job = IBMQJob(self, job_info.get('id'), self._provider._api,
```

Comment:
`IBMQjob` constructor takes a `BaseBackend` as its first parameter. With this change I'm guessing it'll be updated to take a backend name instead of a backend instance? And `job.backend()` will just return the name string.

Reply:
Correct - this goes along the lines of a comment by @luisg5 too, and the first parameter indeed needs to be fixed! It will likely spawn a follow-up to be tackled at `job.backend()` - as in the original document, it was not contemplated what should happen for jobs whose backend does not exist anymore.

Reply:
In the end decided to leave it empty, and tackle it in a follow-up 612e512 when we have a better picture (ie. after refactoring).